### DISCUSSION TITLE
Unable to get the configuration as I'd like

### DISCUSSION THREAD

**Discussion by kipusoep** [renovatebot/renovate#31124] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure, v38.58.0

### Please tell us more about your question or problem

Basically I would like to have 2 package rules:
1. Internal packages
    - They start with our company name, hence `"matchPackagePrefixes": [ "Contoso." ]`
    - Any type of update is allowed; major, minor, patch etc.
    - No need to wait for stability
    - May be updated at any time
2. External packages
    - No major updates are allowed
    - Packages should be at least 7 days old, hence `"stabilityDays": 7`
    - We only update these in the weekends, hence `"extends": [ "schedule:weekends" ]` (we trigger the pipeline automatically every weekend)

But all updates need to be combined together in a single PR, hence `"groupName": "all dependencies"`.  
We're running microservices, so renovate already creates like 20 PRs every weekend, we don't want to have more than 1 per service.

This is my current config (and I tried a lot of variations):
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "enabledManagers": [ "nuget", "dockerfile" ],
  "rebaseWhen": "behind-base-branch",
  "labels": [ "Renovate", "BackEnd" ],
  "internalChecksFilter": "strict",
  "groupName": "all dependencies",
  "separateMajorMinor": false,
  "prHourlyLimit": 0,
  "packageRules": [
    {
      "matchUpdateTypes": [ "major" ],
      "enabled": false
    },
    {
      "matchUpdateTypes": [ "minor", "patch", "pin", "digest" ],
      "stabilityDays": 7,
      "extends": [ "schedule:weekends" ]
    },
    {
      "matchPackagePrefixes": [ "Contoso." ],
      "matchUpdateTypes": [ "major", "minor", "patch", "pin", "digest" ],
      "enabled": true,
      "stabilityDays": 0,
      "schedule": [ "at any time" ]
    }
  ]
}
```

This results in no branch/PR creation, because:
> DEBUG: Checking schedule "every weekend" (repository=Contoso/RenovateTests, branch=renovate/all-dependencies)
       "parsedSchedule": {"schedules": [{"d": [1, 7]}], "exceptions": [], "error": -1}
> DEBUG: Package not scheduled (repository=Contoso/RenovateTests, branch=renovate/all-dependencies)
> DEBUG: Skipping branch creation as not within schedule (repository=Contoso/RenovateTests, branch=renovate/all-dependencies)

I get that for the 2nd package rule, but why doesn't it create anything for the 3rd?

**Comment by rarkins** [comment]
As a start, you cannot have different schedules within the same PR. So decide if you want to have separate groups/PRs or a shared schedule
